Antidote/AbstractCallViewController.m

```diff
@@ -29,85 +30,91 @@ @implementation AbstractCallViewController
-    _callController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall predicate:predicate delegate:self];
-    _callController.delegate = self;
-
-    _call = [[_callController fetchedObjects] firstObject];
+    [super viewDidLoad];
 
-    if (! _call) {
-        return nil;
-    }
+    [self setupBlurredBackground];
 
-    _manager = manager;
+    [self createTopViewContainer];
+    [self createNameLabel];
+    [self createSublabel];
 
-    return self;
+    [self installConstraints];
 }
 
-- (void)viewDidLoad
+- (void)setupBlurredBackground
 {
-    [super viewDidLoad];
-
+    self.view.backgroundColor = [UIColor clearColor];
     self.view.opaque = NO;
 
-    self.view.backgroundColor = [UIColor clearColor];
+    UIVisualEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleDark];
+    UIVisualEffectView *visualEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    visualEffectView.frame = self.view.bounds;
 
-    [self createNameLabel];
+    [self.view insertSubview:visualEffectView atIndex:0];
 
-    [self installConstraints];
+    [visualEffectView makeConstraints:^(MASConstraintMaker *make) {
+        make.edges.equalTo(self.view);
+    }];
 }
 
-- (void)createNameLabel
+- (void)setNickname:(NSString *)nickname
 {
-    if (self.call.chat.friends.count == 1) {
-        OCTFriend *friend = self.call.chat.friends.firstObject;
-
-        self.nameLabel = [UILabel new];
-        self.nameLabel.text = friend.nickname;
-        self.nameLabel.font = [[AppContext sharedContext].appearance fontHelveticaNeueWithSize:30.0];
-        self.nameLabel.textColor = [UIColor whiteColor];
-        self.nameLabel.textAlignment = NSTextAlignmentCenter;
-        [self.nameLabel sizeToFit];
-
-        [self.view addSubview:self.nameLabel];
-    }
+    _nickname = nickname;
+    self.nameLabel.text = nickname;
+    [self.nameLabel setNeedsDisplay];
 }
 
-- (void)installConstraints
+- (void)createTopViewContainer
 {
-    [self.nameLabel makeConstraints:^(MASConstraintMaker *make) {
-        make.top.equalTo(self.view.top).with.offset(kIndent);
-        make.centerX.equalTo(self.view.centerX);
-        make.height.equalTo(30);
-    }];
+    self.topViewContainer = [UIView new];
+    self.topViewContainer.backgroundColor = [UIColor darkGrayColor];
+    self.topViewContainer.alpha = 0.95;
+    [self.view addSubview:self.topViewContainer];
 }
 
-#pragma mark -  RBQFetchedResultsControllerDelegate
-
-- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+- (void)createNameLabel
 {
-    self.call = [[self.callController fetchedObjects] firstObject];
+    self.nameLabel = [UILabel new];
+    self.nameLabel.text = self.nickname;
+    self.nameLabel.font = [[AppContext sharedContext].appearance fontHelveticaNeueWithSize:kNameLabelFontSize];
+    self.nameLabel.textColor = [UIColor whiteColor];
+    self.nameLabel.textAlignment = NSTextAlignmentCenter;
+    [self.nameLabel sizeToFit];
```

Comment:
I don't think that you need to call `sizeToFit`. It should resize automatically with autolayout.